core/src/autogluon/core/trainer/abstract_trainer.py

```diff
@@ -468,12 +521,51 @@ def train_multi_levels(
                 additional_full_weighted_ensemble=additional_full_weighted_ensemble,
             )
             model_names_fit += base_model_names + aux_models
-        if self.model_best is None and len(model_names_fit) != 0:
+        if (self.model_best is None or infer_limit is not None) and len(model_names_fit) != 0:
             self.model_best = self.get_model_best(can_infer=True, infer_limit=infer_limit, infer_limit_as_child=True)
-        self._time_limit = None
+        self._callbacks_conclude()
+        self._fit_cleanup()
         self.save()
         return model_names_fit
 
+    def _fit_setup(self, time_limit: float | None = None, callbacks: List[AbstractCallback] = None):
+        """
+        Prepare the trainer state at the start of / prior to a fit call.
+        Should be paired with a `self._fit_cleanup()` at the conclusion of the fit call.
+        """
+        self._time_train_start = time.time()
+        self._time_train_start_last = self._time_train_start
+        self._time_limit = time_limit
+        self.reset_callbacks()
+        if callbacks is not None:
+            assert isinstance(callbacks, list), f"`callbacks` must be a list. Found invalid type: `{type(callbacks)}`."
```

Comment:
Does it make sense to verify that callbacks are of type `AbstractCallback`?

Reply:
Good idea! Added